sdk/core/azure-core-amqp/samples/internal/eventhub_async_writer_sample/eventhub_async_writer_sample.cpp

```diff
@@ -24,7 +21,7 @@ int main()
   std::string targetEntity = credentials->GetEntityPath();
   if (targetEntity.empty())
   {
-    targetEntity = GetEnvHelper::GetEnv("EVENTHUB_NAME");
```

Comment:
I'd write a handling for `nullptr` being returned.
It can be in a follow-up PR, because I'd also review the samples that use `getenv()` already, and make sure that they do handle it, and come up with a code that is (almost) identical in all our samples, preferably with the same error message and so on.

Reply:
`GetEnvHelper` won't ever return `nullptr`, because it returns a `std::string` and throws if the environment variable isn't present.

That's actually the primary reason for using `GetEnvHelper` instead of `std::getenv` - `GetEnvHelper` resolves several weaknesses in `std;:getenv`.